libpermian/plugins/github/__init__.py

```diff
@@ -0,0 +1,119 @@
+import requests
+import jinja2
+import logging
+import json
+from libpermian.plugins import api
+from libpermian.reportsenders.base import BaseReportSender
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class GitHubReportingException(Exception):
+    def __init__(self, status_code, text):
+        super().__init__(f'Error {status_code}, {text}')
+
+
+@api.reportsenders.register('github-pr')
+class GitHubPullRequestReportSender(BaseReportSender):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        github_token = self.settings.get('github', 'token')
+        self.pr_id = self.settings.get('github', 'pull-request')
+        self.repository = self.settings.get('github', 'repository')
+        self.api_url = 'https://api.github.com'
+        
+        self.headers = {'Accept': 'application/vnd.github.v3+json', \
+                        'Authorization': f'token {github_token}'}
+
+        # Get output text template
+        if self.reporting.data and 'output-template' in self.reporting.data:
+            jinja_env = jinja2.Environment(loader=jinja2.BaseLoader)
+            self.template = jinja_env.from_string(self.reporting.data.get('output-template'))
+        else:
+            jinja_env = jinja2.Environment(loader=jinja2.PackageLoader('libpermian.plugins.github', 'templates'))
+            self.template = jinja_env.get_template('output_text.j2')
+
+    def render_output_text(self):
+        return self.template.render(crcs=self.caseRunConfigurations,
+                                    reportsender=self)
+
+    def get_payload(self, status="in_progress", conclusion=None, head_sha=None):
```

Comment:
The `get_payload` sounds like the payload already exists somewhere and the method extracts it. I'd suggest using different name like `make_payload`, `prepare_payload`, `construct_payload` or `generate_payload` so that it's clear the method creates the payload.